src/modules/rlm_python/rlm_python.c

```diff
@@ -445,46 +442,117 @@ static rlm_rcode_t do_python_single(REQUEST *request, PyObject *pFunc, char cons
 	 *	If request is NULL, pass None.
 	 */
 	tuplelen = 0;
-	if (request != NULL) {
-		for (vp = fr_cursor_init(&cursor, &request->packet->vps);
-		     vp;
-		     vp = fr_cursor_next(&cursor)) tuplelen++;
-	}
+	for (vp = fr_cursor_init(&cursor, vps);
+	     vp;
+	     vp = fr_cursor_next(&cursor)) tuplelen++;
 
-	if (tuplelen == 0) {
-		Py_INCREF(Py_None);
-		pArgs = Py_None;
+	int i = 0;
+	if ((pArg = PyTuple_New(tuplelen)) == NULL) {
+		pArg = NULL;
 	} else {
-		int i = 0;
-		if ((pArgs = PyTuple_New(tuplelen)) == NULL) {
-			ret = RLM_MODULE_FAIL;
-			goto finish;
-		}
-
-		for (vp = fr_cursor_init(&cursor, &request->packet->vps);
-		     vp;
-		     vp = fr_cursor_next(&cursor), i++) {
+		for (vp = fr_cursor_init(&cursor, vps);
+			 vp;
+			 vp = fr_cursor_next(&cursor), i++) {
 			PyObject *pp;
 
 			/* The inside tuple has two only: */
 			if ((pp = PyTuple_New(2)) == NULL) {
+				return -1;
```

Comment:
pArg might not be freed if you return like that. 
Should probably call to Py_XDECREF(pArg)